main.js

```diff
@@ -0,0 +1,275 @@
+/******CONSTANTS*******/
+const Medium = document.getElementById("Medium");
+const Hard = document.getElementById("Hard");
+const Easy = document.getElementById("Easy");
+const highScore = document.getElementById("high-score");
+const Score = document.getElementById("score");
+const speedRadios = document.querySelectorAll('input[name="speed"]');
+
+const canvas = getCanvas();
+const ctx = canvas.getContext("2d");
+let newDirection = "right", oldDirection = "right";
+let v =0;// 
+let food = {x:300,y:300}
+let highscore=0;
+
+let body = [{ x: 300, y: 300, newDirection: "right"}, { x: 270, y: 300, newDirection: "right"}, { x: 240, y: 300, newDirection: "right"}, { x: 210, y: 300, newDirection: "right"}];
+
+
+function getCanvas() {
+    // Select the game canvas
+    const canvas = document.querySelector("#game-canvas");
+    return canvas;
+}
+
+function setCanvasSize() {
+    const parent = canvas.parentNode;
+    canvas.width = parent.offsetWidth;
+    while(canvas.width % 30 !=0){
+        canvas.width++;
+    }
+    canvas.height = parent.offsetHeight;
+    while(canvas.height % 30 !=0){
+        canvas.height++;
+    }
+}
+
+function getCanvasSize() {
+    return {
+        width: canvas.width,
+        height: canvas.height
+    }
+}
+
+function clearScreen() {
+    const { width, height } = getCanvasSize();
+    ctx.clearRect(0, 0, width, height);
+}
+
+function drawSquare(x, y, size, fill, borderWidth = 2, radius = 3) {
+    ctx.fillStyle = fill;
+    ctx.lineWidth = borderWidth;
+    ctx.strokeStyle = 'black';
+
+    // Begin path for rounded rectangle
+    ctx.beginPath();
+    ctx.moveTo(x + radius, y);
+    ctx.lineTo(x + size - radius, y);
+    ctx.quadraticCurveTo(x + size, y, x + size, y + radius);
+    ctx.lineTo(x + size, y + size - radius);
+    ctx.quadraticCurveTo(x + size, y + size, x + size - radius, y + size);
+    ctx.lineTo(x + radius, y + size);
+    ctx.quadraticCurveTo(x, y + size, x, y + size - radius);
+    ctx.lineTo(x, y + radius);
+    ctx.quadraticCurveTo(x, y, x + radius, y);
+    ctx.closePath();
+
+    // Fill the rectangle
+    ctx.fill();
+
+    // Stroke the rectangle border
+    ctx.stroke();
+}
+
+// moving diretion
+
+function move() {
+
+    for (let h = 0; h < body.length; h++) {
+        if (body[h].newDirection === "right") {
+            body[h].x += v;
+
+        } else if (body[h].newDirection === "left") {
+            body[h].x -= v;
+
+        } else if (body[h].newDirection === "up") {
+            body[h].y -= v;
+
+        } else if (body[h].newDirection === "down") {
+            body[h].y += v;
+
+        }
+    }
+
+}
+function restart(){
+    body = [{ x: 300, y: 300, newDirection: "right"}, { x: 270, y: 300, newDirection: "right"}, { x: 240, y: 300, newDirection: "right"}, { x: 210, y: 300, newDirection: "right"}];
```

Comment:
In here you can see again, why having a initialize function `createSnake` is mandatory to have.